Assets/HotmaxEditorVR/Hittables/propTile.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class propTile : MonoBehaviour, IHittable
+{
+    //--local refs
+    private stateManager _stateManagerMutatorRef;
+
+    public int currentPage;
+
+    public GameObject spawnableProp;
+    public TextMesh propTileText;
+    private GameObject boundingScaleBox;
+
+    void Start()
+    {
+        _stateManagerMutatorRef = init._stateManagerMutatorRef;
+        displayPropInMenu(spawnableProp, currentPage);
+    }
+
+    public void receiveHit(RaycastHit hit)
+    {
+        //get the correct object from the current chunk
+        GameObject newProp = Instantiate(spawnableProp);
+        newProp.name = spawnableProp.name;
+        newProp.transform.parent = init.props.transform;
+        newProp.SetActive(true);
+
+        //set the spawned object to the new selected object
+        _stateManagerMutatorRef.SET_SELECTED_OBJECT(newProp);
+        newProp.transform.position = hit.point;
+        //check for prop -some objects are loaded with props, some aren't. 
+        if (!newProp.GetComponent<prop>())
+        {
+            newProp.AddComponent<prop>();
+        }
+        newProp.AddComponent<activeProp>();
+        _stateManagerMutatorRef.SET_SELECTED_OBJECT_IS_ACTIVE(true);
+    }
+
+    //show the prop model on the tile and rotate it around
+    void displayPropInMenu(GameObject menuProp, int page)
+    {
+        float biggestPropSide;
+
+        GameObject menuPropDisplay = Instantiate(menuProp);
+        menuPropDisplay.name = menuProp.name + " (visual)";
+
+        //create a bounding box so the prop can be scaled proportionally
+        boundingScaleBox = GameObject.CreatePrimitive(PrimitiveType.Cube);
+        boundingScaleBox.GetComponent<MeshRenderer>().enabled = false;
+        boundingScaleBox.GetComponent<BoxCollider>().enabled = false;
+        boundingScaleBox.transform.position = transform.position;
+
+        Vector3 menuPropDimension = Vector3.zero;
```

Comment:
group this with the other variable at the top of the function

Reply:
got it